threadpool-isolation/src/main/java/io/shiftleft/threadpoolisolation/StructuredConcurrencyContext.java

```diff
@@ -0,0 +1,97 @@
+package io.shiftleft.threadpoolisolation;
+
+import java.util.concurrent.*;
+
+abstract public class StructuredConcurrencyContext implements AutoCloseable {
```

Comment:
Can you please add some docs for the public api, e.g. a brief "what this class does", differences between enter and enterOnce, usage examples.